eng/pipelines/preview.yml

```diff
@@ -24,11 +32,11 @@ pool:
 extends:
   template: eng/pipelines/templates/stages/archetype-autorest-preview.yml@azure-sdk-tools
   parameters:
-    RegenerationJobCount: 20
+    RegenerationJobCount: 10
```

Comment:
What motivated this change?

Reply:
There was no considerable change in runtime when fanning out so wide.

Compare 20 legs:
![image](https://github.com/Azure/autorest.csharp/assets/1291634/177a235a-4574-4eaa-948a-f8ad76a85e41)

to 10:
![image](https://github.com/Azure/autorest.csharp/assets/1291634/70bea19c-ffbd-405b-b143-1567dfe3fb39)

Reply:
Until we can use a weighted grouping, were stuck with some leg taking about 20 minutes.